src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -306,6 +306,21 @@ jlong ObjectSynchronizer::_last_async_deflation_time_ns = 0;
 static uintx _no_progress_cnt = 0;
 static bool _no_progress_skip_increment = false;
 
+#define CHECK_THROW_NOSYNC_IE(obj) \
+  if (obj->klass()->is_inline_klass()) { \
+    ResourceMark rm(THREAD); \
+    const char* desc = "Cannot synchronize on an instance of value class "; \
+    const char* className = obj->klass()->external_name(); \
+    size_t msglen = strlen(desc) + strlen(className) + 1; \
+    char* message = NEW_RESOURCE_ARRAY(char, msglen); \
```

Comment:
If NEW_RESOURCE_ARRAY fails, it will call vm_exit_out_of_memory, so no need to check for nullptr.

Reply:
Do you need EnableValhalla ?